### PR TITLE
tox: run tests for real for fastapislim

### PR DESCRIPTION
**Pull request by xrmx**
# Description

Add missing entry for running pytest with fastapislim

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [ ] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -903,6 +903,7 @@ commands =
   lint-instrumentation-fastapi: isort --diff --check-only --settings-path {toxinidir}/.isort.cfg {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi
   lint-instrumentation-fastapi: flake8 --config {toxinidir}/.flake8 {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi
   lint-instrumentation-fastapi: sh -c "cd instrumentation && pylint --rcfile ../.pylintrc opentelemetry-instrumentation-fastapi"
+  test-instrumentation-fastapislim: pytest {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/tests {posargs}
```
Comment (emdneto) [inline tox.ini:906]: I'm wondering if we can use the pattern of {0,1} indexes to have some lines removal:

```
--- a/tox.ini
+++ b/tox.ini
@@ -115,10 +115,12 @@ envlist =
     lint-instrumentation-falcon
 
     ; opentelemetry-instrumentation-fastapi
-    py3{8,9,10,11,12}-test-instrumentation-fastapi
-    py3{8,9,10,11,12}-test-instrumentation-fastapislim
-    pypy3-test-instrumentation-fastapi
-    pypy3-test-instrumentation-fastapislim
+    ; The numbers at the end of the environment names
+    ; below mean these dependencies are being used:
+    ; 0: fastapi
+    ; 1: fastapi-slim
+    py3{8,9,10,11,12}-test-instrumentation-fastapi-{0,1}
+    pypy3-test-instrumentation-fastapi-{0,1}
```

Wdyt? With that can remove some lines of pip install

Comment (xrmx) [reply tox.ini:906]: Two things:
- I don't think 0 and 1 are explicit enough for the fact we are testing two different packages
- the diff would require also to remove the pre_commands lines right?

Comment (emdneto) [reply tox.ini:906]: 1. Yes
Something like that:
```
-  fastapi: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements.txt
-  fastapislim: pip install opentelemetry-api@{env:CORE_REPO}\#egg=opentelemetry-api&subdirectory=opentelemetry-api
-  fastapislim: pip install opentelemetry-semantic-conventions@{env:CORE_REPO}\#egg=opentelemetry-semantic-conventions&subdirectory=opentelemetry-semantic-conventions
-  fastapislim: pip install opentelemetry-sdk@{env:CORE_REPO}\#egg=opentelemetry-sdk&subdirectory=opentelemetry-sdk
-  fastapislim: pip install opentelemetry-test-utils@{env:CORE_REPO}\#egg=opentelemetry-test-utils&subdirectory=tests/opentelemetry-test-utils
-  fastapislim: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements-slim.txt
+  fastapi-0: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements-0.txt
+  fastapi-1: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements-1.txt
+  lint-instrumentation-fastapi: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements-0.txt
```

Comment (xrmx) [reply tox.ini]: Updated the PR with a third option :sweat_smile:

File: tox.ini [hunk 2]
```diff
@@ -555,12 +553,8 @@ commands_pre =
...
   fastapi: pip install opentelemetry-test-utils@{env:CORE_REPO}\#egg=opentelemetry-test-utils&subdirectory=tests/opentelemetry-test-utils
-  fastapi: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements.txt
-  fastapislim: pip install opentelemetry-api@{env:CORE_REPO}\#egg=opentelemetry-api&subdirectory=opentelemetry-api
-  fastapislim: pip install opentelemetry-semantic-conventions@{env:CORE_REPO}\#egg=opentelemetry-semantic-conventions&subdirectory=opentelemetry-semantic-conventions
-  fastapislim: pip install opentelemetry-sdk@{env:CORE_REPO}\#egg=opentelemetry-sdk&subdirectory=opentelemetry-sdk
-  fastapislim: pip install opentelemetry-test-utils@{env:CORE_REPO}\#egg=opentelemetry-test-utils&subdirectory=tests/opentelemetry-test-utils
-  fastapislim: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements-slim.txt
+  fastapi-main: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements.txt
+  fastapi-slim: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements-slim.txt
```
Comment (emdneto) [inline tox.ini:557]:
```suggestion
  fastapi-slim: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements-slim.txt
  lint-instrumentation-fastapi: pip install -r {toxinidir}/instrumentation/opentelemetry-instrumentation-fastapi/test-requirements.txt
```

Comment (emdneto) [reply tox.ini:557]: This should fix the lint

**Review by emdneto — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (emdneto) [reply tox.ini]: Seems much better than {0,1} indexes. Thanks

**Pull request merged**